src/builder/create_attachment.rs

```diff
@@ -114,6 +139,39 @@ impl<'a> CreateAttachment<'a> {
     }
 }
 
+/// A wrapper around some base64-encoded image data. Used when an endpoint expects the image
+/// payload directly as part of the JSON body, instead of as a multipart upload.
+#[derive(Clone, Debug, Serialize)]
+#[serde(transparent)]
+pub struct ImageData(String);
+
+impl ImageData {
+    /// Constructs image data from a base64-encoded blob of data. The string must be a valid data
+    /// URI, for example:
+    ///
+    /// ```
+    /// use serenity::builder::ImageData;
+    ///
+    /// let s = "data:image/png;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=";
+    /// assert!(ImageData::from_base64(s).is_ok());
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`Error::Url`] if the string is not a valid data URI. See the [Discord
+    /// docs](https://discord.com/developers/docs/reference#image-data).
+    pub fn from_base64(s: &str) -> Result<Self> {
```

Comment:
Can this be changed to either take `String` or take `impl Into<Cow<'a, str>>`?

Reply:
How come? I would expect any type that implements `Into<Cow<'a, str>>` also implements `AsRef<str>`.

Reply:
Am I missing something? This is taking a borrowed string in, performing a check which is expected to always pass, then cloning the borrowed string into an owned string? I am saying take the owned string originally so you don't have to clone it.

Reply:
Ah sorry, right.